tools/topology/topology2/cavs/cavs-nocodec.conf

```diff
@@ -124,7 +124,7 @@ Object.Dai {
 		# include DAI copier components
 		Object.Widget.copier."0" {
 			index 5
-			dai_index 1
+			dai_index 2
```

Comment:
looks good @yongzhi1 but is dai_index needed at for IPC4?

Reply:
> looks good @yongzhi1 but is dai_index needed at for IPC4?

Not sure about the IPC4 protocol. Without explicit setting, the copier is instantiated with correct dai_index, should I remove all the redundant dai_index here to reduce confusion?

Reply:
yes I think so. I dont think we need it at all @bardliao can you please confirm?

Reply:
> yes I think so. I dont think we need it at all @bardliao can you please confirm?

In theory, we don't need dai_index for IPC4, but we didn't remove the using of dai_index from current linux code base. I don't know if it will cause any issue if we remove it from topology.

Reply:
To clarify, when dai_index was removed from cavs-nocodec.conf, the dai_index is generated correctly when we build the tplg. So dai_index can be used in case we want to overwrite the default value, otherwise we can remove it from top conf level w/o changing the net results:

```
	'copier.SSP.6.1.sof_tkn_dai.word' {
		tokens sof_tkn_dai
		tuples {
			word {
				dai_index 2
				direction 1
			}
		}
	}
```
```
```

Reply:
Let's move it from the top level conf @yongzhi1

Reply:
@yongzhi1 yes, the base class value is used as default if you dont override it later.